Loop/Managers/LoopAppManager.swift

```diff
@@ -368,6 +369,49 @@ class LoopAppManager: NSObject {
         }
         return false
     }
+    
+    private func resetLoop() {
+        deviceDataManager.pumpManager?.prepareForDeactivation({ _ in })
+        
+        resetLoopUserDefaults()
+        resetLoopDocuments()
+    }
+    
+    private func resetLoopUserDefaults() {
+        // Store values to persist
+        let allowDebugFeatures = UserDefaults.appGroup?.allowDebugFeatures
+        let studyProductSelection = UserDefaults.appGroup?.studyProductSelection
```

Comment:
Does `allowSimulators` also need to be restored?

Thinking more about `studyProductSelection` being needed in `TidepoolKit` or `LoopKit` (like you had originally), it is only needed because user default value need to be restored after the wipe. I get that you want to return to a `refresh install` state as part of this wipe without losing these more static values. It seems these static values could only be those that are exposed by `LoopConfigurator`, eh? I was going to suggest just reseting specific elements of the user default, but then you might not reset ones added in the future.

Now that I understand the needs better, I agree with your original plan of putting these details in `LoopKit`. Sorry about the back and forth.

Ideally, I think the code related to study product selection should not be exposed to `Common` repos, since this is a Tidepool only concern. But I cannot think of a clean way to accomplish this.